tools/generators/migrate-converged-pkg/index.ts

```diff
@@ -710,42 +722,108 @@ function setupStorybook(tree: Tree, options: NormalizedSchema) {
 
   return tree;
 }
+/**
+ * Remove function after migration is complete.
+ */
+function migrateE2ESetupToCypress(tree: Tree, options: NormalizedSchema) {
+  const e2ePath = joinPathFragments(options.projectConfig.root, 'e2e');
+  const e2eFolderExists = tree.exists(e2ePath);
 
-function moveStories(tree: Tree, options: NormalizedSchema) {
-  const componentName = names(options.normalizedPkgName).className.replace('React', '');
-  const sourceRoot = options.projectConfig.sourceRoot ?? '';
-  const oldStoriesPath = joinPathFragments(sourceRoot, 'stories');
-  const newStoriesPath = joinPathFragments(oldStoriesPath, componentName);
-  const storiesExistInNewPath = tree.exists(newStoriesPath);
+  if (!e2eFolderExists) {
+    return;
+  }
 
-  if (storiesExistInNewPath) {
+  visitNotIgnoredFiles(tree, e2ePath, treePath => {
+    if (treePath.includes('selectors.ts')) {
+      const newFilePath = joinPathFragments(options.paths.sourceRoot, 'testing', path.basename(treePath));
+
+      // Move testing helper file to src/testing.
+      tree.rename(treePath, newFilePath);
+    } else if (treePath.includes('.e2e.')) {
+      const content = tree.read(treePath)?.toString('utf-8');
+      const fileName = path.basename(treePath).replace('e2e', 'cy');
+      const componentName = fileName.split('.')[0];
+      const newCypressTestPath = joinPathFragments(options.paths.sourceRoot, 'components', componentName, fileName);
+      // Move cypress component test file to appropriate src/components/{ComponentName} location.
+      tree.rename(treePath, newCypressTestPath);
+
+      //Update file imports of cypress component test file.
+      if (content && content.includes('./selectors')) {
```

Comment:
nested if - obvious sign that this should have been a separate function so nesting are mitigated. for this purpose though its not needed as this code will be removed after we migrate probably.

Reply:
Yeah I could have definitely split up this function into multiple ones but opted not to since it'll be removed after migration anyway 😆